detail.py

```diff
@@ -0,0 +1,47 @@
+from django_mako_plus import view_function, jscontext
+from catalog import models as cmod
+
+@view_function
+
+def process_request(request, product:cmod.Product):
+
+  # products = cmod.Product.objects.filter(id= request.dmp.urlparams[0])
```

Comment:
I think if you pass products differently you won't have to use all of these IF statements and define all of the attributes of products again

Reply:
I think the way I set my code up requires me to pass the product objects through the context as a list. But, having less if statements, and slimming down my code would be something that I can work on.